新規会員登録画面UI修正

ippei-shimizu opened the pull request:
## 概要

- 新規会員登録画面のUIを修正しました。スマホ時のみのレイアウトを整えています。
- 「登録ボタン」クリック後のエラーメッセージは、ユーザー操作の妨げにならないように5秒後に消えるようにしました。
-  iOSデバイス時に、inputにfocusすると画面がズームされないように設定しました。

## 確認方法

1. メールアドレス・パスワードを未入力で「登録ボタン」をクリックし、エラーメッセージ表示されて、5秒後に消えます。
2. https://buzzbase-front-git-16-signup-ui-fix-ippei-shimizu.vercel.app/　← こちらの作業ブランチ用検証環境をiOSデバイスでアクセスし、input要素選択時に、画面がズームされないか確認する。（もし、Androidを使用している方でしたら、Androidでも画面がズームされないか確認をお願いしたいです。）

## チェックリスト

- [ ] エラーメッセージが5秒後に消えるか。
- [ ] iOSデバイス時にinput要素を選択し、画面がズームされないか。（iOSデバイスをお持ちの方のみ）
- [ ] Androidデバイス時にinput要素を選択し、画面がズームされないか。（Androidデバイスをお持ちの方のみ）
- [ ] その他、コードの書き方に問題がないか。

## コメント
今回のプルリクはUI修正中心になってしまいますが、もしUIでもっとこうした方がユーザビリティが向上するのではないか、などのご意見がありましたら、ご意見をいただきたいです。

yushi32 commented:
開発お疲れ様です。

iPhoneから記載のリンクにアクセスし、下記の通り確認しました。
- [x] エラーメッセージが5秒後に消えるか。
- [x] iOSデバイス時にinput要素を選択し、画面がズームされないか。（iOSデバイスをお持ちの方のみ）
- [x] その他、コードの書き方に問題がないか。  

エラーメッセージが下の要素に被ってしまう点が少し気になりました。

[![Image from Gyazo](https://i.gyazo.com/e79f3cbb78b43fcfb1f6a65fd3d943af.png)](https://gyazo.com/e79f3cbb78b43fcfb1f6a65fd3d943af)

ippei-shimizu commented:
@yushi32 
レビューありがとうございます！
yushiさんのおっしゃる通り、エラーメッセージが下のフォームに被ってしまうのは、フォーム入力の妨げになってしまう可能性があると思いますので、表示位置を少し上に修正しました。（headerに被るのは、フォーム入力には影響はないかなと思いました）
エラー表示位置をposition: fixed;で指定した理由は、エラーメッセージが表示された時にフォームが移動してしまうのが、よくないのかなと思いました！

↓position: fixed;を使用しなかったイメージ
[![Image from Gyazo](https://i.gyazo.com/cb3bf70b4e91b993305928b050a8dc07.gif)](https://gyazo.com/cb3bf70b4e91b993305928b050a8dc07)

ippei-shimizu commented:
@yushi32 
ご確認ありがとうございます！
ご指摘いただけたおかげで色々なパターンを確認することができました。
エラーメッセージ表示位置を、修正する前の位置に戻しましたので、再度ご確認をお願いいたします！

yushi32 commented:
お疲れ様です。
再度iPhoneから表示を確認いたしました。
LGTMです！

ippei-shimizu commented:
@yushi32 
何度の確認してくださりありがとうございました！
